src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1059,14 +1061,24 @@ void OrbitMainWindow::on_actionOpen_Capture_triggered() {
     return;
   }
 
-  OpenCapture(file.toStdString());
+  const bool starts_with_file_target = (target_configuration_.index() == 2);
+  const bool has_data = app_->HasCaptureData();
+  // If the current OrbitMainWindow starts with a FileTarget and the capture file is cleared, open
+  // new capture file in the current OrbitMainWindow. Otherwise, open capture files in new windows.
+  OpenCapture(file.toStdString(), !starts_with_file_target || has_data);
 }
 
-void OrbitMainWindow::OpenCapture(const std::string& filepath) {
-  app_->OnLoadCapture(filepath);
-  setWindowTitle(QString::fromStdString(filepath));
-  UpdateCaptureStateDependentWidgets();
-  FindParentTabWidget(ui->CaptureTab)->setCurrentWidget(ui->CaptureTab);
+void OrbitMainWindow::OpenCapture(const std::string& filepath, bool open_in_new_window) {
+  if (open_in_new_window) {
+    QProcess::startDetached(QString::fromStdString(orbit_executable_path_),
```

Comment:
If I'm not mistaken this will drop all other command line flags, is this intended? Not sure if we really need to care since the flags are only meant for development...

Reply:
I add a new commit to get the other command line flags when opening a new window. But I'm not sure whether my implementation is a good choice.

I just checked that we can not use the return value of `absl::ParseCommandLine(argc, argv)` . The return value will ignore all arguments passed after the flag-terminating delimiter (`--`), see [here](https://abseil.io/docs/cpp/guides/flags#parsing-flags-during-startup) for more details. Hence, if we start with command line flags like `--devmode`, we cannot get it from the returned value of `absl::ParseCommandLine(argc, argv)` .